### DISCUSSION TITLE
addCustomHeaders() not working on Media Library v10/Laravel 9

### DISCUSSION THREAD

**Discussion by duytanvu** [spatie/laravel-medialibrary#2801] [Q&A] [answered]
### Issue

After upgrading from **Laravel 8 + Media Library v8** to **Laravel 9 + Media Library v10**, files uploaded to S3 using:

addCustomHeaders(['ACL' => 'public-read'])

are no longer assigned the `public-read` ACL.

On Laravel 8 / Media Library v8:
- Files were uploaded with `public-read` ACL
- Accessible via `->getUrl()`

After upgrading:
- Files upload successfully
- ACL is missing → objects are private
- `->getUrl()` no longer works

This appears related to the upgrade to:

league/flysystem-aws-s3-v3 ^3.0

**Comment by duytanvu** [comment, marked as answer]
This seems to be caused by Flysystem v3.

In `createOptionsFromConfig()`, configuration values are appended under the `params` key.  
As a result, the ACL is stored as:

$options['params']['ACL']

However, the adapter later tries to read:

$options['ACL']

Because of this mismatch, the ACL option is ignored during upload.

I've opened a PR to address this.

**Reply by freekmurze** [reply]
Thanks!
